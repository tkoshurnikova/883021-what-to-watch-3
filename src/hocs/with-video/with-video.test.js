import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withVideo from "./with-video.jsx";

const film = {
  title: `The No Flowers`,
  image: `http://placehold.it/280x175`,
  description: `qwerty`,
  rating: 1.0,
  numberOfVotes: 1982,
  director: `Taika Di Caprio`,
  actors: `Ashley Cooper`,
  genre: `Dramas`,
  year: 1900,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withVideo(MockComponent);

it(`withVideo is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      isPlaying={false}
      film={film}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
