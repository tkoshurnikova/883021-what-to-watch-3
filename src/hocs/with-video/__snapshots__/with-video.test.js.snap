// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideo is rendered correctly 1`] = `
<div>
  <video
    autoPlay={true}
    height="175"
    muted={true}
    poster="http://placehold.it/280x175"
    width="280"
  />
</div>
`;
